plugins/main/ptp/ptp.go

```diff
@@ -57,93 +57,115 @@ func setupContainerVeth(netns, ifName string, mtu int, pr *types.Result) (string
 	// "192.168.3.1/32 dev $ifName" and "192.168.3.0/24 via 192.168.3.1 dev $ifName".
 	// In other words we force all traffic to ARP via the gateway except for GW itself.
 
-	var hostVethName string
+	hostInterface := &types.Interface{}
+	containerInterface := &types.Interface{}
+
 	err := ns.WithNetNSPath(netns, func(hostNS ns.NetNS) error {
-		hostVeth, _, err := ip.SetupVeth(ifName, mtu, hostNS)
+		hostVeth, contVeth, err := ip.SetupVeth(ifName, mtu, hostNS)
 		if err != nil {
 			return err
 		}
+		containerInterface.Name = contVeth.Attrs().Name
+		containerInterface.Mac = contVeth.Attrs().HardwareAddr.String()
+		hostInterface.Name = hostVeth.Attrs().Name
+		hostInterface.Mac = hostVeth.Attrs().HardwareAddr.String()
+
+		pr.Interfaces = []*types.Interface{hostInterface, containerInterface}
+		for _, ipc := range pr.IP {
+			// All addresses apply to the container veth interface
+			ipc.Interface = 1
+		}
 
 		if err = ipam.ConfigureIface(ifName, pr); err != nil {
 			return err
 		}
 
-		contVeth, err := netlink.LinkByName(ifName)
+		// Re-fetch container veth to update attributes
+		contVeth, err = netlink.LinkByName(ifName)
 		if err != nil {
 			return fmt.Errorf("failed to look up %q: %v", ifName, err)
 		}
 
-		// Delete the route that was automatically added
-		route := netlink.Route{
-			LinkIndex: contVeth.Attrs().Index,
-			Dst: &net.IPNet{
-				IP:   pr.IP4.IP.IP.Mask(pr.IP4.IP.Mask),
-				Mask: pr.IP4.IP.Mask,
-			},
-			Scope: netlink.SCOPE_NOWHERE,
-		}
-
-		if err := netlink.RouteDel(&route); err != nil {
-			return fmt.Errorf("failed to delete route %v: %v", route, err)
-		}
-
-		for _, r := range []netlink.Route{
-			netlink.Route{
+		for _, ipc := range pr.IP {
+			// Delete the route that was automatically added
+			route := netlink.Route{
 				LinkIndex: contVeth.Attrs().Index,
 				Dst: &net.IPNet{
-					IP:   pr.IP4.Gateway,
-					Mask: net.CIDRMask(32, 32),
+					IP:   ipc.Address.IP.Mask(ipc.Address.Mask),
+					Mask: ipc.Address.Mask,
 				},
-				Scope: netlink.SCOPE_LINK,
-				Src:   pr.IP4.IP.IP,
-			},
-			netlink.Route{
-				LinkIndex: contVeth.Attrs().Index,
-				Dst: &net.IPNet{
-					IP:   pr.IP4.IP.IP.Mask(pr.IP4.IP.Mask),
-					Mask: pr.IP4.IP.Mask,
+				Scope: netlink.SCOPE_NOWHERE,
```

Comment:
The diff just makes it look odd.  We're indenting the route-delete+route-add one level and moving them into a for() loop for all addresses, so you're looking at the route-delete block.  That scope hasn't changed.  A full diff would be clearer.